jina/clients/websockets.py

```diff
@@ -0,0 +1,96 @@
+import asyncio
+from typing import Callable, List
+
+from .base import BaseClient
+from .helper import callback_exec
+from ..importer import ImportExtensions
+from ..logging.profile import TimeContext, ProgressBar
+from ..types.request import Request
+
+if False:
+    from ..types.request import Response
+
+
+class WebSocketBaseClient(BaseClient):
+    async def _get_results(self,
```

Comment:
can we split this into smaller functions?

Reply:
Since this works directly inside context managers & async for loops (and a lot of comments), I feel, it is at the right level of modularization. I'll take a look at the gRPC client & try to see if something else can be refactored.